bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/discovery/FreeboxOsDiscoveryService.java

```diff
@@ -107,49 +107,50 @@ protected void startScan() {
                 List<LanHost> lanHosts = new ArrayList<>(thingHandler.getManager(LanBrowserManager.class).getHosts()
                         .stream().filter(LanHost::reachable).toList());
 
-                discoverServer(thingHandler.getManager(SystemManager.class), bridgeUID);
-                discoverPhone(thingHandler.getManager(PhoneManager.class), bridgeUID);
-                discoverPlugs(thingHandler.getManager(FreeplugManager.class), bridgeUID);
-                discoverRepeater(thingHandler.getManager(RepeaterManager.class), bridgeUID, lanHosts);
-                discoverPlayer(thingHandler.getManager(PlayerManager.class), bridgeUID, lanHosts);
-                discoverVM(thingHandler.getManager(VmManager.class), bridgeUID, lanHosts);
-                discoverHome(thingHandler.getManager(HomeManager.class), bridgeUID);
+                discoverServer(bridgeUID);
+                discoverPhone(bridgeUID);
+                discoverPlugs(bridgeUID);
+                discoverRepeater(bridgeUID, lanHosts);
+                discoverPlayer(bridgeUID, lanHosts);
+                discoverVM(bridgeUID, lanHosts);
+                discoverHome(bridgeUID);
                 if (thingHandler.getConfiguration().discoverNetDevice) {
-                    discoverHosts(thingHandler, bridgeUID, lanHosts);
+                    discoverHosts(bridgeUID, lanHosts);
                 }
             } catch (FreeboxException e) {
                 logger.warn("Error while requesting data for things discovery: {}", e.getMessage());
             }
         }
     }
 
-    private void discoverHome(HomeManager homeManager, ThingUID bridgeUID) throws FreeboxException {
+    private void discoverHome(ThingUID bridgeUID) throws FreeboxException {
```

Comment:
The exception is caught within the method itself and not re-thrown, so it can be removed from the method signature.
```suggestion
    private void discoverHome(ThingUID bridgeUID) {
```

Reply:
Fixed